jormungandr/src/network/mod.rs

```diff
@@ -352,6 +352,11 @@ fn connect_and_propagate_with<F>(
         }
     };
     let node_id = node.id();
+    assert_ne!(
```

Comment:
```suggestion
    debug_assert_ne!
```

what about a debug assert instead?

and returning early with an error in the log if that is the case? Just would like to avoid crash now on the node for something that we can _work around_ if it happens

Reply:
Could do, but this is meant to just double-check after poldercast which should not give us the self node in views. I think I'd try to go in, make a fix in poldercast and update Cargo.toml with the new version for this branch.

Reply:
@NicolasDP is already reworking poldercast, replaced this with a warning and an early return until that is done.